packages/eip1193/src/index.ts

```diff
@@ -42,39 +42,30 @@ export class EIP1193 extends Connector {
     })
   }
 
-  /** {@inheritdoc Connector.connectEagerly} */
-  public async connectEagerly(): Promise<void> {
+  private async connect(eager: boolean): Promise<void> {
     const cancelActivation = this.actions.startActivation()
 
-    return Promise.all([
-      this.provider.request({ method: 'eth_chainId' }) as Promise<string>,
-      this.provider.request({ method: 'eth_accounts' }) as Promise<string[]>,
-    ])
-      .then(([chainId, accounts]) => {
-        this.actions.update({ chainId: parseChainId(chainId), accounts })
-      })
-      .catch((error) => {
+    try {
+      // Wallets may resolve eth_chainId and hang on eth_accounts pending user interaction, which may include changing
+      // chains; they should be requested serially, with accounts first, so that the chainId can settle.
+      const accounts = await (eager ? this.provider.request({ method: 'eth_accounts' }) : this.provider
```

Comment:
Question about this `eager` flag. I don't see it being used before? We were running `eth_accounts` in both instances. Is this change intentional?

Reply:
Ok, I see this has to do with `eager` that can be `true` or `false`, depending on the call-site. Is there a better way to describe its intent? Is it something like connect if already authorised vs explicitly authorise? In that case, why fallback to `eth_accounts` when `eth_requestAccounts` failed? This is probably a bit out of scope for this PR, but a good opportunity to expand my `web3-react` knowledge a bit!

Reply:
I believe this is done because EIP1193 does not necessarily support eth_requestAccounts, so a fallback is necessary. You don't see this in other connectors because they are not explicitly EIP1193 connectors.

The codepaths for connectEagerly and activate were identical except for the usage of eth_accounts vs eth_requestAccounts, so I added connect as a shared method to express that with less code duplication.

I've refactored to `activateAccounts(requestAccounts: () => Promise<string[]>)`.